lms/templates/help_modal.html

```diff
@@ -149,11 +158,106 @@
+                        courseName,
+                        courseId,
+                        option,
+                        defaultOptionText = '${_("- Select -") | n, js_escaped_string}',
+                        markedSelectedOption = false;
+
+                    // Make sure courseOptions is empty before we begin pushing options into it.
+                    courseOptions = [];
+
+                    for (i = 0; i < data.length; i++) {
+                        courseDetails = data[i]['course_details'];
+                        if (!courseDetails) {
+                            continue;
+                        }
+
+                        courseName = courseDetails['course_name'];
+                        courseId = courseDetails['course_id'];
+                        if (!(courseName && courseId)) {
+                            continue;
+                        }
+
+                        // Build an option for this course and select it if it's the course we're currently viewing.
+                        if (!markedSelectedOption && courseId === currentCourseId) {
+                            option = buildCourseOption(courseName, courseId, true);
+                            markedSelectedOption = true;
+                        } else {
+                            option = buildCourseOption(courseName, courseId, false);
+                        }
+
+                        courseOptions.push(option);
+                    }
+
+                    // Build the default option and select it if we haven't already selected another option.
+                    if (markedSelectedOption) {
+                        option = buildCourseOption(defaultOptionText, '', false);
+                    } else {
+                        option = buildCourseOption(defaultOptionText, '', true);
+                    }
+
+                    // Add the default select option to the head of the courseOptions Array.
+                    courseOptions.unshift(option);
+
+                    finishedLoadingCourseOptions = true;
+                },
+                complete: function() {
+                    courseOptionsLoadInProgress = false;
+                }
+            });
+        },
+        buildCourseOption = function(courseName, courseId, selected) {
+            var option = '<option value="' + _.escape(courseId) + '"';
+            if (selected) {
+                option += ' selected';
+            }
+            option += '>' + _.escape(courseName) + '</option>';
+            return option;
         };
 
+    % if user.is_authenticated():
+        userAuthenticated = true;
+    % endif
+
+    % if course:
+        currentCourseId = "${unicode(course.id)}";
+    % endif
+
     DialogTabControls.setKeydownListener($helpModal, $closeButton);
 
     $(".help-tab").click(function() {
+        if (userAuthenticated && !finishedLoadingCourseOptions && !courseOptionsLoadInProgress) {
+            loadCourseOptions();
+        }
```

Comment:
I talked with product about the best way to handle a delay in the response from the server. Ultimately, it was decided that if we were to kick off the request immediately when the user clicks on the help tab, we should have enough time to get a response before they click the second link to render a specific form. In the event that we haven't received a response by then, we will show the form without the course select.